### PR TITLE
RANK: Optimize OTSU filter

### PR DESCRIPTION
**Pull request by hmaarrfk**
I simplified the computation of OTSU by making most things multiplications. Gives a ~5x~ 4x speedup on random data.

<details> <summary> ASV is weird, I never know how to interpret the benchmarks</summary>

```
Optimizing OTSU:

Initial benchmark
otsu              2.57±0.09ms     145±2ms

No normalization 0be1e7718da91a85adc2e173dc4ecdff2f52f321
otsu              1.13±0.07ms     65.0±2ms

Even more simplification 31eeabd37aadab35fe70a624b29890485c1b0647
otsu              887±40μs    46.2±0.7ms

OTSU: skip unneded computations.
otsu              678±40μs     30.4±2ms


OTSU: keeping everything pop a double (two tries for the benchmark)
otsu               723±40μs     37.8±6ms
otsu              842±200μs    42.9±10ms

```
</details>

Redoing the benchmarks without touching my computer

```
[  0.00%] · For scikit-image commit c40b04b0 <optimize_otsu>:
                           otsu              514±0.6μs   25.2±0.02ms 
[ 20.00%] · For scikit-image commit 10c4c0ff <optimize_otsu~1>:
                           otsu               757±10μs    41.0±0.1ms 
[ 40.00%] · For scikit-image commit 89602d89 <optimize_otsu~2>:
                           otsu               934±5μs    52.4±0.07ms 
[ 60.00%] · For scikit-image commit cfa8855c <optimize_otsu~3>:
                           otsu              1.96±0.01ms    119±0.3ms  
[ 80.00%] · For scikit-image commit 5f03ec55 <optimize_otsu~4>:
                           otsu              1.95±0.01ms    119±0.2ms  
```

From what I remember, this needs https://github.com/scikit-image/scikit-image/pull/3506 for the tests to pass.

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/filters/rank/generic_cy.pyx [hunk 1]
```diff
@@ -8,11 +8,11 @@ from libc.math cimport log, exp
 
 from .core_cy cimport dtype_t, dtype_t_out, _core
 
-from ..._shared.interpolation cimport round 
+from ..._shared.interpolation cimport round
```
[the remark this answers is not in the document]
Comment (hmaarrfk) [reply skimage/filters/rank/generic_cy.pyx]: I really wanted to, but libc round is so much slower....

**Comment by hmaarrfk** [comment]
@ahojnnes I scaled back the changes not included the `double pop` -> `integer pop`. Honestly, I think using double is a mistake, but it is probably best left to an other PR.

**Comment by hmaarrfk** [comment]
`otsu~4` is like master. The one that doesn't change was due to the double->Py_ssize_t change that I reverted.  

Master is about (2ms, 120 ms), this PR is at (500us, 25ms) for small and large images.
To get asv to output this, you need to do:

1. Delete the gigantic matrix of builds that are currently benched otherwise it will take for ever.
2. `asv run master..head -b RankSuite`

In this case, it is going to to build 4 or 5 envs, and run them in **slow** mode.

Fast mode isn't super reliable sometime. I know many libraries are change averse when the output changes given a certain input, so I wanted to make sure I portrayed the gains correctly. On my computer, the output seems to have changed now for an other input that happaned to be a pathological edge case.

**Comment by jni** [comment]
@hmaarrfk good stuff!

I don't particularly want fast mode, I just want to use my current environment by default. Perhaps a PR to asv is warranted...

**Comment by hmaarrfk** [comment]
yeah, using -E existing is super useful. But sometimes doing a real benchmark is useful. I'm torn on benchmarks.

**Pull request merged**